retry.go

```diff
@@ -161,15 +161,23 @@ type clientInterface interface {
 }
 
 type retryHTTP struct {
-	ctx      context.Context
-	client   clientInterface
-	req      requestFunc
-	method   string
-	fullURL  *url.URL
-	headers  map[string]string
-	body     []byte
-	timeout  time.Duration
-	raise4XX bool
+	ctx         context.Context
+	client      clientInterface
+	req         requestFunc
+	method      string
+	fullURL     *url.URL
+	headers     map[string]string
+	body        []byte
+	bodyCreator func() ([]byte, error)
+	timeout     time.Duration
+	raise4XX    bool
+}
+
+func (r *retryHTTP) createBody() ([]byte, error) {
+	if r.bodyCreator != nil {
+		return r.bodyCreator()
+	}
+	return r.body, nil
```

Comment:
can bodyCreator be actually nil? Because from what I can see, either you set it explicitly or set it in `setBody`

Reply:
You are right. I removed that function at all.